modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/DefaultXmlDslElementModelConverter.java

```diff
@@ -153,65 +119,34 @@ private void writeApplicationElement(Element element, DslElementModel<?> element
-                                       ComponentConfiguration config) {
-    Optional<String> defaultValue = getDefaultValue(name, model);
-    if (!defaultValue.isPresent() || !defaultValue.get().equals(config.getValue().get())) {
-      element.setTextContent(config.getValue().get());
-      if (parentNode != element) {
-        parentNode.appendChild(element);
+  private void setTextContentElement(Element element, DslElementModel<?> elementModel, Element parentNode) {
+    getCustomizedValue(elementModel).ifPresent(value -> {
+      if (elementModel.getDsl().supportsChildDeclaration() && !elementModel.getDsl().supportsAttributeDeclaration()) {
+        element.setTextContent(value);
+        if (parentNode != element) {
+          parentNode.appendChild(element);
+        }
+      } else {
+        parentNode.setAttribute(elementModel.getDsl().getAttributeName(), value);
       }
-    }
-  }
-
-  private boolean hasPopulatedParameters(ComponentConfiguration config) {
-    return !(config.getParameters().isEmpty() && config.getNestedComponents().isEmpty());
-  }
-
-  private void populateTypeBasedOnDsl(Element element, ComponentConfiguration config, DslElementSyntax dsl,
-                                      Object model, Element parentNode) {
-
-    final MetadataType type = model instanceof ObjectType ? (MetadataType) model : ((Typed) model).getType();
-
-    dsl.getAttributes()
-        .forEach(attr -> getExplicitParameter(attr.getAttributeName(), config, type)
-            .ifPresent(value -> element.setAttribute(attr.getAttributeName(), value)));
-
-    if (isTextContent(config)) {
-      addTextContentParameter(element, dsl.getAttributeName(), type, parentNode, config);
-      return;
-    }
-
-    config.getNestedComponents().forEach(component -> {
-      ComponentIdentifier identifier = component.getIdentifier();
-      Optional<DslElementSyntax> childDsl = findMatchingDsl(dsl.getChilds(), identifier);
-      if (!childDsl.isPresent()) {
-        childDsl = findMatchingDsl(dsl.getGenerics().values(), identifier);
-      }
-
-      childDsl.ifPresent(innerDsl -> {
-        Element childElement = createElement(innerDsl);
-        populateTypeBasedOnDsl(childElement, component, innerDsl, type, element);
-      });
     });
+  }
 
-    if (parentNode != element) {
-      parentNode.appendChild(element);
+  private Optional<String> getCustomizedValue(DslElementModel elementModel) {
+    Optional<String> defaultValue = getDefaultValue(elementModel.getDsl().getAttributeName(), elementModel.getModel());
+    String value = (String) elementModel.getValue().get();
+    if (!defaultValue.isPresent() || !defaultValue.get().equals(value)) {
+      return Optional.of(value);
     }
+    return empty();
   }
 
-  private boolean isTextContent(ComponentConfiguration config) {
-    return config.getNestedComponents().isEmpty() && config.getValue().isPresent();
-  }
+  private Optional<String> getDefaultValue(String name, Object model) {
```

Comment:
it can be an ObjectFieldType if we are representing a complex type like a message builder, then `body` doesn't have a parameter model associated but a field instead